grafana/data_source_amixr_escalation_chain.go

```diff
@@ -0,0 +1,57 @@
+package grafana
+
+import (
+	"errors"
+	"fmt"
+
+	amixrAPI "github.com/grafana/amixr-api-go-client"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func DataSourceEscalationChain() *schema.Resource {
+	return &schema.Resource{
+		Description: `
+* [HTTP API](https://grafana.com/docs/grafana-cloud/oncall/oncall-api-reference/escalation_chains/)
+`,
+		Read: dataSourceEscalationChainRead,
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The escalation chain name.",
+			},
+		},
+	}
+}
+
+func dataSourceEscalationChainRead(d *schema.ResourceData, m interface{}) error {
+	client := m.(*client).amixrAPI
+	if client == nil {
+		err := errors.New("amixr api client is not configured")
+		return err
+	}
+	options := &amixrAPI.ListEscalationChainOptions{}
+	nameData := d.Get("name").(string)
+
+	options.Name = nameData
+
+	escalationChainsResponse, _, err := client.EscalationChains.ListEscalationChains(options)
+
+	if err != nil {
+		return err
+	}
```

Comment:
Nit: error checking should be immediately after the action that might trigger an error is.
```suggestion
	escalationChainsResponse, _, err := client.EscalationChains.ListEscalationChains(options)
	if err != nil {
		return err
	}
```